File: app/src/main/java/abdullah/elamien/bloodbank/viewmodels/ShoutoutsViewModel.java
package abdullah.elamien.bloodbank.viewmodels;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

import java.util.List;
import java.util.Map;

import javax.inject.Inject;

import abdullah.elamien.bloodbank.models.Shoutout;
import abdullah.elamien.bloodbank.repository.ShoutoutRepository;

/**
 * Created by AbdullahAtta on 6/18/2019.
 */
public class ShoutoutsViewModel extends ViewModel {

    private ShoutoutRepository mRepository;

    @Inject
    public ShoutoutsViewModel(ShoutoutRepository repository) {
        this.mRepository = repository;
    }

    public void createShoutout(Map<String, Object> shoutout) {
        mRepository.createShoutout(shoutout);
    }

    public LiveData<List<Shoutout>> getShoutouts() {
        return mRepository.getShoutouts();
    }

}

File: app/src/main/java/abdullah/elamien/bloodbank/viewmodels/ShoutoutsViewModel.kt
package abdullah.elamien.bloodbank.viewmodels

import abdullah.elamien.bloodbank.models.Shoutout
import abdullah.elamien.bloodbank.repository.ShoutoutRepository
import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import org.koin.core.KoinComponent
import org.koin.core.inject

/**
 * Created by AbdullahAtta on 6/18/2019.
 */
class ShoutoutsViewModel : ViewModel(), KoinComponent {
    private val mRepository: ShoutoutRepository by inject()

    val shoutouts: LiveData<List<Shoutout>>?
        get() = mRepository.shoutouts

    fun createShoutout(shoutout: Map<String, Any>) {
        mRepository.createShoutout(shoutout)
    }

}
